### PR TITLE
Validate globalIP annotation for Pods when using globalnet

### PR DESCRIPTION
**Pull request by sridhargaddam**
In this PR, we validate the globalIP annotation on the connectorPod
as part of e2e tests.

Partially fixes# https://github.com/submariner-io/submariner/issues/544

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: test/e2e/framework/pods.go [hunk 1]
```diff
@@ -42,3 +43,24 @@ func (f *Framework) DeletePod(cluster ClusterIndex, podName string, namespace st
 		return nil, KubeClients[cluster].CoreV1().Pods(namespace).Delete(podName, &metav1.DeleteOptions{})
 	}, NoopCheckResult)
 }
+
+// AwaitPodByAnnotation queries the Pod and looks for the presence of annotation.
+func (f *Framework) AwaitPodByAnnotation(cluster ClusterIndex, annotation string, podName string, namespace string) *v1.Pod {
```
Comment (vthapar) [inline test/e2e/framework/pods.go:48]: AwaitUntilAnnotationOnPod() sounds like better name. AwaitPodByAnnotation() suggests we're waiting for multiple pods to be spawned with annotation used as a filter.

Comment (mangelajo) [reply test/e2e/framework/pods.go:48]: AwaitUntilAnnotationOnPod sounds like the pod is pre-existing and you are only waiting for the annotation.

Comment (mangelajo) [reply test/e2e/framework/pods.go:48]: Which is probably what we wanted, lol

Comment (mangelajo) [reply test/e2e/framework/pods.go:48]: Well but the code is waiting for a pod to appear, and the annotation to be on the pod.

What about : AwaitPodWithAnnotation ?

Comment (sridhargaddam) [reply test/e2e/framework/pods.go]: Mmm.. I used the current name to be aligned with the similar name we used for Service. But I agree the name is confusing and I'm a bit inclined to @vthapar's suggestion - i.e., AwaitUntilAnnotationOnPod.
Also changing the name of the service to keep them in sync.

**Pull request merged**